01-JS-DOM-APIs/js/javascript.js

```diff
@@ -87,3 +87,107 @@ for(let rep of results){     /*iterate and show the elements in html*/
 }
 
 }
+
+
+function criptoTable() {
+const cryptoURL="https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=100&page=1&sparkline=true&price_change_percentage=24h%22%20-H%20%22accept:%20application/json";
+
+fetch(cryptoURL)
+.then(data => data.json())
+.then( data =>{
+
+    tableGenerator(data);
+    
+})
+.catch(data =>{
+
+    console.log("error detected");
+})
+
+
+}
+
+
+function tableGenerator(data){
+
+    const crypto=data;
+
+    const getTable = document.getElementById("table"); //select the section
+
+    const table =document.createElement("table"); //create the table
+
+    const headtable=document.createElement("tr"); //create the head of the table
+
+    const headName=document.createElement("th");    //assign the articles in the table header
+    const headSymbol=document.createElement("th");
+    const headCurrentPrice=document.createElement("th");
+    const headMarketCapRank=document.createElement("th");
+    const headAth=document.createElement("th");
+
+    headName.appendChild(document.createTextNode("name"));
+    headSymbol.appendChild(document.createTextNode("Symbol"));
+    headCurrentPrice.appendChild(document.createTextNode("Current Price"));
+    headMarketCapRank.appendChild(document.createTextNode("Market Carp Rank"));
+    headAth.appendChild(document.createTextNode("All Time High(ath)"));
+
+    headtable.appendChild(headName);
+    headtable.appendChild(headSymbol);
+    headtable.appendChild(headCurrentPrice);
+    headtable.appendChild(headMarketCapRank);
+    headtable.appendChild(headAth);
```

Comment:
You could improve  this function by making small functions that return the parent document after the append of the children, so in this case for example headTable, and the same applies to the body.